content/cloud-computing/concepts/paas/paas.md

```diff
@@ -0,0 +1,54 @@
+---
+Title: 'PaaS'
+Description: 'Platform as a service (PaaS): a service model that offers developers a platform to create, manage, and deploy their applications.'
+Subjects:
+  - 'Computer Science'
+  - 'Networking'
+Tags:
+  - 'Cloud'
+  - 'Platform'
+  - 'Development'
+  - 'Infrastructure'
+CatalogContent:
+  - 'foundations-of-cloud-computing'
+  - 'paths/back-end-engineer-career-path'
+---
+
+## Platform as a Service
+
+**Platform as a service (PaaS)** is a service model offered by cloud service providers that allows developers to develop and deploy apps without worrying about the underlying software and hardware their applications depend upon. When using a PaaS, the service provider provides and maintains the following:
+
+- Operating systems
+- Software development kits (SDK)
+- Frameworks
+- Libraries
+- APIs
+- Any other relevant software needed for application development
+
+In addition to the software, the service provider also provides the infrastructure needed for the application. PaaS includes the services provided by an infrastructure-as-a-service (IaaS) offering. Compared to the other two cloud services available - IaaS and software as a service (SaaS) - PaaS sits right in the middle, placing more responsibility on the service provider than IaaS and less responsibility than SaaS.
+
+## Advantages of PaaS
+
+Using a PaaS is often cheaper than building and maintaining the necessary software and hardware in-house. Many vendors charge based on a pay-per-use model in which a developer is billed only for the resources they use. If a developer were to design and maintain their platform instead of using a PaaS, they would risk incorrectly estimating how much of a resource they need leading them to pay for unused resources. PaaS providers handle scaling.
+
+A cloud service such as PaaS allows a team of developers to collaborate on a project regardless of each developer's geographic location.
+
+Since a PaaS relieves developers of many maintenance tasks, an application can be developed and deployed much more quickly.
+
+## Disadvantages of PaaS
+
+When using a PaaS, developers are limited to the use of the software technologies (langauges, frameworks, libraries, etc...) that are available from the vendor. This is known as vendor lock-in. Before selecting a vendor for PaaS, developers should make sure that they are comfortable using what the vendor provides. Another disadvantage of PaaS is migration. It is possible for a vendor's technology stack to not be compatible with another vendor's technology stack; therefore if a team of developers wishes to change vendor, they may be required to completely rework everything for things to work on the new vendor's platform.
```

Comment:
"When using a PaaS, developers are limited to the use of the software technologies (langauges, frameworks, libraries, etc...) that are available from the vendor. This is known as vendor lock-in."

This is not vendor lock-in. However, PaaS does cause vendor lock-in. Vendor lock-in is a situation in which it is very difficult to move away from a cloud service provider because your application is so dependent on their services. This is covered by the second part of the paragraph, which describes the actual vendor lock-in.